### ISSUE TITLE
New-User: Error, if ExternalId or ExternalType are not defined

### ISSUE THREAD

**Issue by pbimpisidis** [dfensgmbh/biz.dfch.PS.Appclusive.Client#105]
ExternalId and ExternalType are not mandatory parameters for New-User cmdlet. They have a default value on Set-User but not on New-User.
So, when you call New-User without defining those fields (e.g. New-ApcUser -Svc $svc -Name $name -Mail $mail;) , then you get error:

```
System.Management.Automation.ParameterBindingArgumentTransformationException: Cannot process
argument transformation on parameter 'ExternalId'. Cannot convert value "" to type
"System.Guid". Error: "Unrecognized Guid format." --->
```

Suggestion : adding the default values [$ExternalId = [guid]::NewGuid(), $ExternalType = 'Internal'] for those parameters on New-User.ps1 too.

**Comment by pbimpisidis** [comment]
One more point for user cmdlets: create/update two users with the same combination of external id and external type is allowed right now. If it shouldn't be allowed there should be an error handling on Set-User and New-User scripts

**Comment by pbimpisidis** [comment]
@seilc1 can you take a look and let me know if I should go on and fix those?

**Comment by rufer7** [comment]
@pbimpisidis 
`ExternalId`:
`$ExternalId` parameter has to be changed to be mandatory and has to be validated to not be null or whitespace (PoSH annotation). Furthermore `$ExternalId` has to be a string instead of a [guid] as the backend accepts strings. Defining a default value is not a good idea.

`ExternalType`: 
`$ExternalId` parameter has to be changed to be mandatory and has to be validated to not be null or whitespace (PoSH annotation). 

@pbimpisidis @stevenpi will fix that bug.

**Comment by rufer7** [comment]
Fixed by PR #107
